### ISSUE TITLE
macosx backend doesn't support numeric fontweight

### ISSUE THREAD

**Issue by fcolas** [matplotlib/matplotlib#3330]
When providing a TrueType-style fontweight value, the macosx backend raise a TypeError:

```python
import matplotlib.pyplot as plt
fig = plt.figure()
fig.text(.5, .5, 'a', fontweight=700)
plt.show()
```
```
Traceback (most recent call last):
  File "/usr/local/Cellar/python/2.7.8/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/matplotlib-1.4.x-py2.7-macosx-10.9-x86_64.egg/matplotlib/artist.py", line 59, in draw_wrapper
    draw(artist, renderer, *args, **kwargs)
  File "/usr/local/Cellar/python/2.7.8/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/matplotlib-1.4.x-py2.7-macosx-10.9-x86_64.egg/matplotlib/figure.py", line 1079, in draw
    func(*args)
  File "/usr/local/Cellar/python/2.7.8/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/matplotlib-1.4.x-py2.7-macosx-10.9-x86_64.egg/matplotlib/artist.py", line 59, in draw_wrapper
    draw(artist, renderer, *args, **kwargs)
  File "/usr/local/Cellar/python/2.7.8/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/matplotlib-1.4.x-py2.7-macosx-10.9-x86_64.egg/matplotlib/text.py", line 538, in draw
    bbox, info, descent = self._get_layout(renderer)
  File "/usr/local/Cellar/python/2.7.8/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/matplotlib-1.4.x-py2.7-macosx-10.9-x86_64.egg/matplotlib/text.py", line 311, in _get_layout
    ismath=False)
  File "/usr/local/Cellar/python/2.7.8/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/matplotlib-1.4.x-py2.7-macosx-10.9-x86_64.egg/matplotlib/backends/backend_macosx.py", line 166, in get_text_width_height_descent
    six.text_type(s), family, size, weight, style)
TypeError: must be string, not int
```

Whereas providing a string representation works as expected:
```python
import matplotlib.pyplot as plt
fig = plt.figure()
fig.text(.5, .5, 'a', fontweight='bold')
plt.show()
```
(I didn't get this issue under Ubuntu, but I wouldn't know which backend was used.)

**Comment by tacaswell** [comment]
Closed by #4113

**Comment by tacaswell** [comment]
@fcolas please ping to have this re-opened if I am confused.
